parquet-column/src/test/java/org/apache/parquet/column/impl/TestColumnWriterV1.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.parquet.column.impl;
+
+import org.apache.parquet.column.ColumnDescriptor;
+import org.apache.parquet.column.ParquetProperties;
+import org.apache.parquet.column.page.PageWriter;
+import org.apache.parquet.schema.PrimitiveType;
+import org.apache.parquet.schema.PrimitiveType.PrimitiveTypeName;
+import org.apache.parquet.schema.Type.Repetition;
+import org.junit.Test;
+
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.anyInt;
+import static org.mockito.Mockito.atLeastOnce;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+public class TestColumnWriterV1 {
+
+  @Test
+  public void testEstimateNextSizeCheckOnZeroUsedMem() throws Exception {
```

Comment:
Does the new test case fail in the master branch without your fix?

Reply:
The new test case is not applicable in master branch as later code has significantly refactored. After refactor ColumnWriterV1 is no longer responsible to slice and flush pages itself. Other components will call its "writePage" method instead.

In 0.10 and prior, ColumnWriterV1 is responsible for slicing pages where this bug resides and this test targets.

Reply:
Thanks for the explanation!

IMHO, we may not have any plan to make a release based on the legacy version. Please advise if I have missed anything. @shangxinli

Reply:
@shangxinli @wgtmac Since this is only applicable to legacy that has no more releases, should we abandon this PR?

Reply:
Yes, I think it does not make sense to fix something that will not be released.

Reply:
Closed, thanks